build.gradle.kts

```diff
@@ -69,3 +77,26 @@ tasks.jacocoTestReport {
     }
     executionData.setFrom(fileTree(projectDir).include("*.exec"))
 }
+
+detekt {
+    buildUponDefaultConfig = true
+    allRules = false
+    config = files("$projectDir/config/detekt.yml")
+    baseline = file("$projectDir/config/baseline.xml")
+    val detektFileTree = fileTree("$projectDir")
+    detektFileTree.exclude("**/build/**").exclude("**/node_modules/**").exclude(".gradle")
+    source = files(detektFileTree)
+}
+
+tasks.withType<io.gitlab.arturbosch.detekt.Detekt>().configureEach {
+    reports {
+        html.required.set(true)
+        xml.required.set(true)
+        txt.required.set(true)
+        sarif.required.set(true)
+    }
+    jvmTarget = java.sourceCompatibility.toString()
+}
+tasks.withType<io.gitlab.arturbosch.detekt.DetektCreateBaselineTask>().configureEach {
+    jvmTarget = java.sourceCompatibility.toString()
+}
```

Comment:
Ist dies an irgendeiner Stelle in SonarCube zu sehen, dass die Reports von Detekt kommen oder wird das da einfach in einem riesen Report mit rein gefuddelt?

Reply:
Sieht man nicht in Sonarqube - aber man sieht Probleme daran, dass die CI failen wird.